src/Form/MessageSettingsForm.php

```diff
@@ -132,15 +128,15 @@ public function buildForm(array $form, FormStateInterface $form_state) {
    * {@inheritdoc}
    */
   public function submitForm(array &$form, FormStateInterface $form_state) {
-    $config = $this->config('message.settings');
+    parent::submitForm($form, $form_state);
+
+    $config = $this->configFactory->getEditable('message.settings');
```

Comment:
Too true, my friend. Too true.

Anyway, it is not working the same as the OG form as your config name in `getEditableConfigNames` on this form is not correct, it's `messag.message` needs to be `message.settings` :) With that you will be able to switch back to `config()`.